src/shared/components/gateway-section/toggle-button/index.js

```diff
@@ -44,6 +45,19 @@ class ToggleButton extends Component {
   handleToggleButton = () => {
     this.props.onClick()
   }
+
+  getToggleText = () => {
+    const { isActive, inProgress, intl: { messages } } = this.props
+
+    if (!isActive) {
+      if (!inProgress) {
+        return messages.serviceWorker.toggleText
+      } else {
+        return messages.serviceWorker.activatingToggleText
```

Comment:
You can simplify the the above returns by a single one with a ternary condition.